rdp: upstream coding style changes

ManMower opened the pull request:
Try to bring RAIL code closer to upstream style.

Signed-off-by: Derek Foreman <derek.foreman@collabora.com>

ManMower committed:
rdp: upstream coding style changes

Try to bring RAIL code closer to upstream style.

Signed-off-by: Derek Foreman <derek.foreman@collabora.com>

hideyukn88 commented on libweston/backend-rdp/rdprail.c:
```diff
@@ -59,163 +59,174 @@ static void rdp_rail_destroy_window(struct wl_listener *listener, void *data);
-		RdpPeerContext *peerCtx = (RdpPeerContext *)client->context; \
-		struct rdp_backend *b = peerCtx->rdpBackend; \
+		freerdp_peer *client = (context)->custom; \
+		RdpPeerContext *peer_ctx = (RdpPeerContext *)client->context; \
```
Just would like to double check, the use of "peerCtx" is from original weston, and still is at upstream https://gitlab.freedesktop.org/wayland/weston/-/blob/main/libweston/backend-rdp/rdp.c#L1506.

ManMower replied:
Yep, rdp.c has been doing this, and shouldn't be (the rest of the project does_variables_with_underscores instead). Making the existing code be consistent with the rest of weston would be nice, but it would also impose a burden on anyone carrying out of tree patches. We'd like newly landed code to be more conformant though.

hideyukn88 commented on libweston/backend-rdp/rdprail.c:
```diff
@@ -2959,136 +3228,145 @@ rdp_rail_peer_activate(freerdp_peer* client)
-		b->use_rdpapplist) {
-		peerCtx->applist_server_context = b->rdpapplist_server_context_new(peerCtx->vcm);
-		if (!peerCtx->applist_server_context)
+	if (api && b->rdprail_shell_name && b->use_rdpapplist) {
```
Is "api" declared ?

ManMower replied:
No, I missed that here - can I just elide the check for api though? nothing appears to use it in this block.

ManMower replied:
Have removed it for now - will put it back if that wasn't a good idea.

hideyukn88 commented on libweston/backend-rdp/rdprail.c:
```diff
@@ -4164,68 +4539,68 @@ rdp_rail_notify_app_list(void *rdp_backend, struct weston_rdprail_app_list_data
+		del_provider.flags = RDPAPPLIST_FIELD_PROVIDER;
+		if (app_list_data->appProvider &&
+		    utf8_string_to_rail_string(app_list_data->appProvider, &del_provider.appListProviderName))
+			applist_ctx->DeleteApplicationListProvider(applistctx, &del_provider);
```
Shouldn't be "applist_ctx" ?

ManMower replied:
Correct. Sorry, I'm building against upstream FreeRDP and don't have applist support, so didn't get a build failure. I'll push a fix shortly.

hideyukn88 commented on libweston/backend-rdp/rdprail.c:
```diff
@@ -4164,68 +4539,68 @@ rdp_rail_notify_app_list(void *rdp_backend, struct weston_rdprail_app_list_data
-			utf8_string_to_rail_string(app_list_data->appGroup, &deleteAppList.appGroup)) {
+		if (app_list_data->appGroup &&
+		    utf8_string_to_rail_string(app_list_data->appGroup, &delete_app_list.appGroup)) {
 			deleteAppList.flags |= RDPAPPLIST_FIELD_GROUP;
```
"delete_app_list.flags" ?

hideyukn88 commented on libweston/backend-rdp/rdprail.c:
```diff
@@ -4235,42 +4610,35 @@ rdp_rail_notify_app_list(void *rdp_backend, struct weston_rdprail_app_list_data
-				goto Exit_updatePath;
-			char *src = (char *)pixman_image_get_data(app_list_data->appIcon);
+			iconData.iconBits = xmalloc(iconData.iconBitsLength);
+			char *src = pixman_image_get_data(app_list_data->appIcon);
```
pixman_image_get_data returns uint32_t *, so don't we need cast to avoid warning?

ManMower replied:
Really wishing I hadn't touched any of the applist code. :)
Correct, again.

ManMower committed:
rdp: Fix build breaks in applist for sloppy style fixups

ManMower committed:
rdp: Remove check for rail shell api

I broke this check in a prior commit, but nothing this block uses it
so instead of fixing the check, I'll remove the check.
